devito/iteration.py

```diff
@@ -1,38 +1,50 @@
 from collections import Iterable
+from itertools import chain
 
 import cgen
-from devito.codeprinter import ccode
+
+from devito.expression import Expression
+from devito.tools import filter_ordered
 
 __all__ = ['Iteration']
 
 
-class Iteration(object):
+class Iteration(Expression):
```

Comment:
And yes, `IterationSpace` sounds ok too. Although I kind of see the `Dimension` as defining the iteration space, since it will eventually know about PML padding and buffering, while the `Iteration` represents a concrete loop in the code tree/list. But again, nomenclature is totally open for debate.